### PR TITLE
Removed semicolon output in critical state

### LINKED ISSUE #55386: Using radio prefixes (";" for example) will not talk to the radio while in critical.

### PR DESCRIPTION
**Pull request by drongood12**
## About The Pull Request
removed semicolon display in critical health

Fixes #55386 

## Why It's Good For The Game

Bug fixes always make the game better

## Changelog
:cl:
fix: Semicolon will no longer display in critical condition
/:cl:

### CODE REVIEW

File: code/modules/mob/mob_say.dm [hunk 1]
```diff
@@ -124,6 +124,8 @@
 			mods[MODE_SING] = TRUE
 		else if(key == ";" && !mods[MODE_HEADSET] && stat == CONSCIOUS)
 			mods[MODE_HEADSET] = TRUE
+		else if(key == ";" && !mods[MODE_HEADSET] && stat != CONSCIOUS)
+			chop_to = length(key) + 1
```
Comment (LemonInTheDark) [inline code/modules/mob/mob_say.dm:128]: This isn't strictly needed mind, chop_to is already set to 2, so as long as one of the else statements passes it'll go through properly.

It'd be more concise to do something like check for the key and if headset is set, then do the stat check, but that obscures the intent of the check in the first place, so perhaps not.

~~man my old code is shit~~

Comment (drongood12) [reply code/modules/mob/mob_say.dm:128]: All the conditions that are found here take into account only the channels of the departments and they have the removal of the channel prefix, but at the same time for the general channel there is no condition under which the semicolon will be removed when a person is in a critical condition, so this was done, of course it is possible not the best implementation

Comment (Ghommie) [inline code/modules/mob/mob_say.dm:128]:
```suggestion
		else if(key == ";" && !mods[MODE_HEADSET])
			if(stat == CONSCIOUS) //necessary indentation so it gets stripped of the semicolon anyway.
				mods[MODE_HEADSET] = TRUE
```

Comment (LemonInTheDark) [reply code/modules/mob/mob_say.dm:128]: This is about what I meant, and on second thought is a bit more clean.

**Pull request merged**